### ISSUE TITLE
some graphs never fill in

### ISSUE THREAD

**Issue by bengland2** [ceph/cephmetrics#77]
some of the grafana displays never fill in, and it's reallly not obvious why.  See [here for example](https://s3.amazonaws.com/ben-england/ceph/Screenshot+from+2017-07-17+18-03-02.png).  Could it state why there is no data if it doesn't display a graph?  ceph -s shows cluster is healthy, and has been up for over 1 hour.  Some of the graphs have red triangle in them and when I hover over that, it just says internal server error.

**Comment by pcuzner** [comment]
@bengland2 if the graphs are empty, the data is not getting to graphite. If you open a browser to the monitoring host at port 80 you should see the graphite web interface. Can you check there that there are metrics. If there is nothing listed we need to look at the collectors.

**Comment by bengland2** [comment]
It was port 8080, not port 80, but ok, I opened it, there are a lot of metrics, but not seeing data in any of them yet.  So suspecting collectd.

**Comment by bengland2** [comment]
Paul, thanks a lot, it's working better, really cool data.  Problems that remain:

- ceph - rados graph shows all zeroes
- OSD latency graph screen gets 503 internal server error.

if you think its appropriate I'll create separate issues for these.

**Comment by pcuzner** [comment]
@bengland2 The reason for your empty rados data is empty data coming from the monitor's admin socket. When I did a perf dump the cluster counters for osd_bytes num_pgs were all zero. Since that's the source of the data, the charts will be blank. Could you update to 12.1.1 and see if the counters are populated.

As for the OSD latency graph, this is down to bluestore. I need to confirm the specific counters that can serve as health indicators for bluestore - once this is done, the osd collector can be updated to gather the stats.  Until then, you only get the os level metrics for the block and block.wal devices. I'll raise an RFE to track bluestore support

**Comment by bengland2** [comment]
ok, I upgraded everything on those hosts, rebooted, and ceph-mgr did not come back on its own.  But when I got that back up, I observed  12.1.1 installed on monitors and non-zero counters:

    # ceph daemon /var/run/ceph/ceph-mon*asok perf dump

    "cluster": {
        "num_mon": 3,
        "num_mon_quorum": 3,
        "num_osd": 8,
        "num_osd_up": 8,
        "num_osd_in": 8,
        "osd_epoch": 133,
        "osd_bytes": 4302709751808,
        "osd_bytes_used": 82346868736,
        "osd_bytes_avail": 4220362883072,
        "num_pool": 9,
        "num_pg": 544,
        "num_pg_active_clean": 544,
        "num_pg_active": 544,
        "num_pg_peering": 0,
       ...

But still nothing from Ceph RADOS dashboard.  Let me know if there's something else I can do.

**Comment by pcuzner** [comment]
@bengland2 yep we moved on to the next challenge with luminous :smile: 

ceph health command no longer gives all the data that I get from jewel/kraken ... so I needed to refactor the health code to account for the lack of information.

I also had to turn of selinux - it was blocking the osd collector accessing the files on the osd mount points.

Anyway take a look now. metrics are now showing - apart from the osd latencies, which is a bluestore thing..There are a number of dashboard fixes in my local branch too, which fix some niggles and layout issues - hoping to get those out tomorrow (7/21)

**Comment by pcuzner** [comment]
@bengland2 is this still an issue for you, or can this be closed?

**Comment by bengland2** [comment]
unfortunately I just pulled latest cephmetrics tonite and it is actually giving me less info than before :-(  This is with bluestore, I'll try it again with filestore and see if that's better.  Luminous version ceph-common-12.1.2-0.el7.x86_64.

**Comment by pcuzner** [comment]
has selinux been turned back on?
is ceph running with mon_health_preluminous_compat=true in ceph.conf?

**Comment by bengland2** [comment]
selinux was in permissive mode, I'm disabling it and will try again later, update here.
ceph is not running with mon_health_preluminous_compat=true.   Did you want that set or not?

**Comment by pcuzner** [comment]
yes - have that value set for the cluster on Luminous. On Luminous the default from ceph health is pretty light on detail - with this setting, the collector can see more. Can you also check the collectd daemon is ok - at least on the mon's?

**Comment by bengland2** [comment]
one reason the playbook doesn't work for bluestore is a set of errors like this:

TASK [ceph-collectd : Restore SELinux context of OSD journals] *****************
skipping: [ip-172-31-39-180] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
fatal: [ip-172-31-55-42]: FAILED! => {"changed": true, "cmd": "restorecon -R -v /var/lib/ceph/osd/*/journal", "delta": "0:00:00.008823", "end": "2017-08-14 14:54:58.602849", "failed": true, "rc": 255, "start": "2017-08-14 14:54:58.594026", "stderr": "restorecon:  lstat(/var/lib/ceph/osd/*/journal) failed:  No such file or directory", "stdout": "", "stdout_lines": [], "warnings": []}
...

Looking at bluestore OSD directories, there is no "journal" softlink, unlike filestore.  Instead there is a block.db and block.wal softlinks.

/var/lib/ceph/osd/ceph-5:
...
lrwxrwxrwx. 1 ceph ceph  58 Aug 14 14:47 block.db -> /dev/disk/by-partuuid/04a572b4-aad5-48c5-96c5-96a5e8fe9981
...
lrwxrwxrwx. 1 ceph ceph  58 Aug 14 14:47 block.wal -> /dev/disk/by-partuuid/22701995-be9f-4eae-bc43-f1799fcc5f14
...

So the ansible task fails and all subsequent tasks are skipped for these nodes, which are all the OSD nodes.  Why is it trying to do this to the journal file?    I'll try skipping this step and see what happens.

**Comment by bengland2** [comment]
I tried skipping the step in .-1 and now all the nodes are visible, data is coming in fine, really great.  Watching a cluster boot in grafana is nice demo.    Why is it necessary to restorecon on the journal softlink in the first place?

**Comment by zmc** [comment]
@bengland2 @pcuzner Just want to close the loop on this - is it resolved or is the above question something that needs to be answered?

**Comment by bengland2** [comment]
sorry, I haven't got an answer quite yet, have had to switch over to linode from AWS but am there now, will try to get back to you in 1 week with this, but close it if you know you fixed it.

**Comment by usmanakramb** [comment]
@bengland2 have you had any luck with getting the graphs to work? I am using 13.2.1 and having similar issues where some graphs are not showing. I have selinux in permissive mode on all the nodes.
